### PR TITLE
Add installation info metadata support

### PR DESCRIPTION
**Pull request by kbogtob**
### What does this PR do?

It adds the installation info metadata to the chef cookbook so the agent returns that it was installed by the triple chef/chef version/cookbook version

### Motivation.

Allow customers to track their chef installation. Allows us to make stats on installation methods.

### CODE REVIEW

File: recipes/install_info.rb [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+  else
+    owner 'dd-agent'
+    mode '600'
+  end
+
+  source 'install_info.yaml.erb'
+
+  variables(
+    install_method: "chef-#{Chef::VERSION}",
+    install_method_version: "datadog-cookbook-#{Chef::Datadog.cookbook_version(run_context)}"
```
Comment (albertvaka) [inline recipes/install_info.rb:54]: Maybe we want to call it `datadog_cookbook` so the dash `-` separates name from version.

File: templates/default/install_info.yaml.erb [hunk 2]
```diff
@@ -0,0 +1,6 @@
+<%= JSON.parse(({
```
Comment (albertvaka) [inline templates/default/install_info.yaml.erb:1]: `JSON.parse().to_json.to_yaml` is this the way to print a yaml? 🤔

File: recipes/install_info.rb (hunk 1 above)
Comment (kbogtob) [reply recipes/install_info.rb:54]: Woops! Good catch

File: templates/default/install_info.yaml.erb (hunk 2 above)
Comment (kbogtob) [reply templates/default/install_info.yaml.erb:1]: C/p from other template, I'll improve

**Pull request merged**